src/config/mod.rs

```diff
@@ -313,10 +313,13 @@ impl Resource {
         // Port with unspecified address will bind to all network interfaces
         // so we have to check for all Port resources if they share the same
         // port.
-        for (key, port, protocol0) in unspecified {
+        for (key, address0, protocol0) in unspecified {
             for (resource, components) in resource_map.iter_mut() {
                 if let Resource::Port(address, protocol) = resource {
-                    if address.port() == port && &protocol0 == protocol {
+                    // IP addresses can either be v4 or v6.
+                    // Therefore we check if the ip version matches, the port matches and if the protocol (TCP/UDP) matches
+                    // when checking for equality.
+                    if address.port() == address0.port() && &protocol0 == protocol && address0.is_ipv4() == address.is_ipv4() {
```

Comment:
Could we just rely on `SocketAddr`'s `Eq` implementation here? I.e. `address == address0 && &protocol = protocol0`?

Reply:
Oh yes! Made the change!

Reply:
Oh sorry my bad (I'm having a few issues with my dev environment) so I'm a bit slow. So looks like we may not be able to rely on the socketAddr implementation of equality because for unspecified ip addresses, it binds to all the ports. I'm guessing what that meant is that we cannot have 0.0.0.0:8080 and 10.10.10.10:8080, because the former will also bind to 10.10.10.10. Should I revert the change back to checking for ip_version?